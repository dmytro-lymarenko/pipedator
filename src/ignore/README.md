```typescript
import { ignore } from 'pipedator';
// or
import { ignore } from 'pipedator/lib/ignore';

```
An alias for [`success`](#success)