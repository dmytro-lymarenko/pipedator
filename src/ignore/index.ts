import { alternative } from '../alternative';

export const ignore = alternative;
