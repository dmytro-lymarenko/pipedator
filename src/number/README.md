- **Description**

Creates a validator to check if a value is a number or not.

- **How to import**

```typescript
import { number } from 'pipedator';
// or
import { number } from 'pipedator/lib/number';

```
- **Signature**

```typescript
function number(message?: string): Validator;
```
- **Parameters**

  - `message` - (optional) custom message.


- **Usage**

```typescript
number().validate(5); // valid
number().validate(NaN); // valid
number().validate('Some text'); // invalid
number('Custom message').validate(null); // invalid with 'Custom message'
```