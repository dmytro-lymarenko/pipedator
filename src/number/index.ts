import { test } from '../test';

export function number<ValidValue = any>(message?: string) {
	return test<ValidValue>(value => typeof value === 'number', message || 'Value should be a number');
}
