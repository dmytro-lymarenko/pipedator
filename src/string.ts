import { createValidator } from './core';

export function string() {
	return createValidator({
		message: 'Value should be a string',
		validate: (value, ctx) => (typeof value === 'string' ? null : { value, message: ctx.message, path: ctx.path }),
	});
}

export function empty() {
	return createValidator({
		message: 'Value should be empty',
		validate: (value, ctx) => (value.length === 0 ? null : { value, message: ctx.message, path: ctx.path }),
	});
}
