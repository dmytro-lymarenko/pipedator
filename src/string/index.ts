import { test } from '../test';

export function string<ValidValue = any>(message?: string) {
	return test<ValidValue>(value => typeof value === 'string', message || 'Value should be a string');
}
