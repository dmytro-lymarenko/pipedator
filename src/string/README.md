- **Description**

Creates a validator to check if a value is a string or not.

- **How to import**

```typescript
import { string } from 'pipedator';
// or
import { string } from 'pipedator/lib/string';

```
- **Signature**

```typescript
function string<ValidValue = any>(message?: string): Validator<ValidValue>;
```
- **Parameters**

  - `message` - (optional) custom message.


- **Usage**

```typescript
string().validate('Some text'); // valid
string().validate(5); // invalid
string('Custom message').validate(null); // invalid with 'Custom message'
```