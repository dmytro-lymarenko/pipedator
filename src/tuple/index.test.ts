import { tuple } from './index';
import { number } from '../number';
import { string } from '../string';
import { ignore } from '../ignore';

describe('tuple()', () => {
	it('should validate', () => {
		expect(tuple([number(), string()]).validate([1, 'text'])).toBeNull();
		expect(tuple([number(), string()]).validate([1, 'text', {}])).toBeNull();
		expect(tuple([ignore(), number()]).validate([null /* or any other value */, 5])).toBeNull();
	});

	it('should fail', () => {
		expect(tuple([number(), string()]).validate(['1', 'text'])).toMatchSnapshot();
		expect(tuple([number(), string()]).validate([1])).toMatchSnapshot();
	});

	it('should fail and return all errors', () => {
		expect(tuple([number(), string()]).validate(['1'])).toMatchSnapshot();
	});

	it('should fail and return only the first error', () => {
		expect(tuple([number(), string()], { onlyFirstError: true }).validate(['1'])).toMatchSnapshot();
	});

	it('should support custom message', () => {
		expect(tuple([number(), string()], undefined, 'Custom message').validate([1])).toMatchSnapshot();
	});
});
