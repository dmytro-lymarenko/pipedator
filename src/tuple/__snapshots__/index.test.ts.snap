// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tuple() should fail 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Array [],
      "message": "Value should be a number",
      "path": Array [
        "0",
      ],
      "value": "1",
    },
  ],
  "message": "Value should have a valid shape",
  "path": Array [],
  "value": Array [
    "1",
    "text",
  ],
}
`;

exports[`tuple() should fail 2`] = `
Object {
  "errors": Array [
    Object {
      "errors": Array [],
      "message": "Value should be a string",
      "path": Array [
        "1",
      ],
      "value": undefined,
    },
  ],
  "message": "Value should have a valid shape",
  "path": Array [],
  "value": Array [
    1,
  ],
}
`;

exports[`tuple() should fail and return all errors 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Array [],
      "message": "Value should be a number",
      "path": Array [
        "0",
      ],
      "value": "1",
    },
    Object {
      "errors": Array [],
      "message": "Value should be a string",
      "path": Array [
        "1",
      ],
      "value": undefined,
    },
  ],
  "message": "Value should have a valid shape",
  "path": Array [],
  "value": Array [
    "1",
  ],
}
`;

exports[`tuple() should fail and return only the first error 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Array [],
      "message": "Value should be a number",
      "path": Array [
        "0",
      ],
      "value": "1",
    },
  ],
  "message": "Value should have a valid shape",
  "path": Array [],
  "value": Array [
    "1",
  ],
}
`;

exports[`tuple() should support custom message 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Array [],
      "message": "Value should be a string",
      "path": Array [
        "1",
      ],
      "value": undefined,
    },
  ],
  "message": "Custom message",
  "path": Array [],
  "value": Array [
    1,
  ],
}
`;
